services/app-web/src/pages/SubmissionSideNav/RateSummarySideNav.tsx

```diff
@@ -0,0 +1,133 @@
+import styles from './SubmissionSideNav.module.scss'
+import { GridContainer, Icon, SideNav } from '@trussworks/react-uswds'
+import {
+    generatePath,
+    matchPath,
+    Outlet,
+    Navigate,
+    useLocation,
+    useParams,
+} from 'react-router-dom'
+import { useAuth } from '../../contexts/AuthContext'
+import { getRouteName } from '../../routeHelpers'
+import { useFetchRateQuery } from '../../gen/gqlClient'
+import { ApolloError } from '@apollo/client'
+import { handleApolloError } from '../../gqlHelpers/apolloErrors'
+import { Error404 } from '../Errors/Error404Page'
+import { recordJSException } from '../../otelHelpers'
+import { GenericErrorPage } from '../Errors/GenericErrorPage'
+import { Loading, NavLinkWithLogging } from '../../components'
+import { RoutesRecord } from '../../constants'
+
+export const RateSummarySideNav = () => {
+    const { id } = useParams() as { id: string }
+    const { loggedInUser } = useAuth()
+    const { pathname } = useLocation()
+    const routeName = getRouteName(pathname)
+
+    const shouldRedirect = matchPath(
+        RoutesRecord.SUBMISSIONS_RATE_QUESTIONS_AND_ANSWERS,
```

Comment:
@JasonLin0991 I agree this is a little funky, especially I am not expecting a redirect to other page inside the side nav for a route already in the sidenav. We are setting up the route just to redirect hehe. 

Don't know the solution without pulling down code more, I can take a look though on Monday. Assuming from you PR description comment you tried[ `<Redirect />`](https://v5.reactrouter.com/web/api/Redirect) already in AppRoutes? Is the correct?

Reply:
With that said, no blocking, you can merge if you feel ready

Reply:
I believe <Redirect /> was replaced with <Navigate />. The complication is the difference in the route paths and parameters between submission summary and rate summary. Summary page `id` param is the contract id where as rate summary it is the rate ID. So we would have had to add more logic in `AppRoutes` to handle getting the rate ID from the url to generate the redirect path. I decided to put this logic in the `RateSummarySideNav` since both routes are nested in this outlet.

I think if we refactored how we setup the routes to use `createBrowserRouter`, maybe [loader](https://reactrouter.com/en/main/route/loader) can do the redirecting.